Rotator controller

zisi opened the pull request:
New features:
* Power supply in the same board
* Better grounding scheme
* Better RS-485 transceiver
* I2C multiplexer
* Temperature sensor
* Use standard size board
* Filters for end stops (de-bouncing)
* Filter in input of 5V regulator
* Use Kicad library convention

It remains:
* Better silkscreen placement
* Remove unused components
* Update existing boards (like encoder) with updated library parts
* PCB review

Commit:
Update satnogs library.

Add new footprints, 3d shapes and schematic components.

Commit:
Updates in rotator controller.

Add RJ45 for communication and power in (PoE).
New dimensions of board (euroboard 100x80 mm)
Embed the power supply to the board (WIP).
Add pins for indicator LED and for rf switch (optional).
Add additional pads for power in case that PoE (RJ45) isn't suitable.
Updates in schematic and board wuth new library parts.

Commit:
Add PCA9540B schematic component, I2C multiplexer.

Commit:
Add fuse holder and stitching via.

zisi committed:
Updates in PCB and schematic.

Add digital and power ground planes (WIP).
Connection of separate GND with zero resistor.
Add I2C multiplexer.
Finalize power supply of 12V.
Change RJ45 to headers.
Add RC filter (LPF) for end-stops.
Add more dev-pins.
Add filter in uC voltage input.

Signed-off-by: zisi <agzisim@gmail.com>

zisi committed:
Add GND-GNDD vias and silkscreen.

Signed-off-by: zisi <agzisim@gmail.com>

zisi committed:
Updates according to Kicad library convention

Signed-off-by: zisi <agzisim@gmail.com>

zisi committed:
Resolve issues #22, #23, #24, #25

Signed-off-by: zisi <agzisim@gmail.com>

ppapadeas committed:
Fix silkscreen placement and zone cleanup

ppapadeas commented:
Fixed some silkscreen placement and edited some zones. Please check commit here https://github.com/ppapadeas/satnogs-rotator-controller/commit/2d7ba4d5a91d47f02779edaf345907a5b1710d78

zisi committed:
Use one GND bottom plane and GND areas in front

Remove zero ohm resistors
To avoid connection with GNDD directly with motor GND use keep out area

Signed-off-by: zisi <agzisim@gmail.com>

ppapadeas commented:
@zisi Anything left for me to review?

zisi commented:
More details about thermal design of power supply, #28.

zisi committed:
Add electrolytic capacitor, TVS-diode in PSU input

Capacitor reduces the voltage ripple of input voltage (long cable)
TVS-diode protect the PSU and all the board from surges and ESD events

Signed-off-by: zisi <agzisim@gmail.com>
